### PR TITLE
INSERT IGNORE data conversion behavior

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
This PR addresses behavior for INSERT IGNORE with regards to data conversion errors.

1) Changes column value to 0 values on incorrect types
2) Truncates strings that are too long

cc. https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html#sql-mode-strict

### CODE REVIEW

File: sql/plan/insert.go [hunk 1]
```diff
@@ -432,6 +474,24 @@ func (i *insertIter) ignoreOrClose(ctx *sql.Context, row sql.Row, err error) (sq
...
+// Per MySQL docs "Rows set to values that would cause data conversion errors are set to the closest valid values instead"
+// cc. https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html#sql-mode-strict
+func (i *insertIter) convertDataAndIgnore(ctx *sql.Context, row sql.Row, columnIdx int, err error) (sql.Row, error) {
+	if sql.ErrLengthBeyondLimit.Is(err) {
+		maxLength := i.schema[columnIdx].Type.(sql.StringType).MaxCharacterLength()
+		row[columnIdx] = row[columnIdx].(string)[:maxLength] // truncate string
+	} else if sql.ErrInvalidValue.Is(err) {
+		row[columnIdx] = i.schema[columnIdx].Type.Zero()
+	} else {
+		return i.ignoreOrClose(ctx, row, err)
```
Comment (zachmu) [inline sql/plan/insert.go:487]: Don't use this here: the semantics are to return a nil row in some cases, which you definitely do not want from this function. Just return an error and let the caller figure out what to do

File: enginetest/insert_queries.go [hunk 2]
```diff
@@ -1167,38 +1167,110 @@ var InsertIgnoreScripts = []ScriptTest{
 		},
 	},
 	{
-		Name: "Test that INSERT IGNORE INTO works with unique keys",
+		Name: "Test that INSERT IGNORE properly addresses data conversion",
```
Comment (zachmu) [inline enginetest/insert_queries.go:1170]: You need a lot more tests of this behavior with lots of different types and values. I would start with unit tests, then move on to engine tests

**Review by zachmu — APPROVED** [review]
Overall approach seems fine, just a couple comments. This needs a lot more tests, and be sure to compare to actual mysql.

File: enginetest/insert_queries.go (hunk 2 above)
Comment (AmirAktify) [reply enginetest/insert_queries.go:1190]: Zach, thank you for this comment. I appreciate the attention to extra robustness in the fix, and that you added tests that make it unlikely to break again.

**Pull request merged**